src/main/java/net/minestom/server/entity/Entity.java

```diff
@@ -504,6 +510,7 @@ public synchronized void switchEntityType(@NotNull EntityType entityType) {
         EntitySpawnType type = entityType.registry().spawnType();
         this.aerodynamics = aerodynamics.withAirResistance(type == EntitySpawnType.LIVING ||
                 type == EntitySpawnType.PLAYER ? 0.91 : 0.98, 1 - entityType.registry().drag());
+        initCollisions();
```

Comment:
name this `updateCollisions` since you are calling it multiple times

Reply:
done